completing person router and handler

ImedZnd opened the pull request:
adding fraud and unfraud queue
fixing the names of the queues in tests
fixing dateDTO
adding message bad person id
deleting useless comma in configuration
changing hardcoded queue names and adding fraud and unfraud in database repository
change the person dto to use the other dtos
refactor the person handler
changing hardcoded queue names and adding fraud and unfraud in queue initializer
fixing typo in PersonQueueListenerTest.kt
adding fraud and unfraud methods in repository
adding test for fraud and unfraud

ImedZnd committed:
starting the branshing

ImedZnd committed:
complete testing person repository

ImedZnd committed:
Merge pull request #1

merging person test repository

ImedZnd committed:
person service and person service test

ImedZnd committed:
Merge pull request #2

person service

ImedZnd committed:
change the repository to return Optional
fixing the repository tests
change the country dto to the extension function
change the country handler to the optional return
change the repository to the optional return
fix the fraud per country route
adding the CountryRouterTest.kt
change the country service to the optional return
creating the CountryServiceTest.kt
create dtos
creating the i18N messages
linting the person model
change the person configuration with the i18N
linting perosndao
refactor person database repository
uncomment the after all method in person database repo
creating the person handler
deleteing two useless method in person repo
creating the person router and tests
delete to string in PersonsByCountry.kt
refactor the methods with annotation before each and all
refactor person test

ImedZnd committed:
adding events to application.yml
refactor the tests with the id
adding the id to the model
adding the rabbittemplate and message queue setting to the database repository
creating person queue initializer and queue setting
change the reactive database to the uuid id
adding save/update/delete/flag method and publish method to the repository
adding find person by id in person service and save/update/delete/flag
adding queue test in service
adding rabbitmq and spring cloud stream dependencies to POM
creating RabbitMQConfiguration.kt
change the sql schema person id to UUID

ImedZnd committed:
returning back to the id Long

ImedZnd committed:
adding the queue listener in application.yml
adding messages for the added return
adding PersonQueueHandler.kt in PersonConfiguration.kt
refactor the PersonDatabaseRepository.kt for more detailed errors
reorder the params in personDtO
add to person method in personDTO
adding get by id method in person rest handler and save/update/delete/flags methods
create sealed interface for generec errors in person repository
adding save/update/delete/flag/id method to the person rest router
adding more tests for the added methods

ImedZnd committed:
deleting unused class
optimize import in PersonHandler.kt
deleting print in PersonQueueHandler.kt
adding test in person router tests

ImedZnd committed:
creating pipeline test analyse and push to docker hub

ImedZnd committed:
fix path

ImedZnd committed:
adding fraud and unfraud queue
fixing the names of the queues in tests
fixing dateDTO
adding message bad person id
deleting useless comma in configuration
changing hardcoded queue names and adding fraud and unfraud in database repository
change the person dto to use the other dtos
refactor the person handler
changing hardcoded queue names and adding fraud and unfraud in queue initializer
fixing typo in PersonQueueListenerTest.kt
adding fraud and unfraud methods in repository
adding test for fraud and unfraud

omarbenromdhane reviewed (changes requested):
a huge pull request is not recommanded , since it will take a lt of time for the person who will review the code, and it will bring a lot of changes
remove 140 println s
what code 3 in country means? pick a name that give reflection to the business model
optimize the code by removing the return type when it is obvious by type inference
remove the Persistable interface from the dao 
remove the name builder from companion objects if it is not building block
no need to use flow of to create server response from single value 
use functions call pipeline instead of wrapping a call of function inside another call of function
using sql for repository means that your microservice is doing more than one thing which is an anti pattern
the usage of the boolean parameter for findAllIsFraudstersByCountry is not clear
don't call fields "database" in service, since service can recieve any implementation of the repository
DateIsNotValidError is not tested
use data class for dtos and daos
do not use @Autowired
person database repository need to be refactored, functions are too complicated
use single line expression function instead of using return when it is a single line function
catch the exception for messageSource.getMessage(string, null, Locale.US)
why to have a dto to wrap person id only and year only ?
use @componen instead of @service for PersonQueueInitializer
do not use @repository, spring will detect it automatically
remove mono from PersonRepository
why to use long and not int for results that will be always an int like count ?
PersonAlreadyFraudsterRepositoryError not tested
use method references for PersonRouter
setting all routes in same function will not allow you to integrate swagger description
add swagger for all µservices
YearNotValidError not tested

ImedZnd committed:
changing the code 3 to code iso in country to reflect better the business model
adding person service to the country service and message source to the country handler
changing the code  to code  iso in country dao
deleting useless method from country repository
deleting useless test from country repository test
changing the code 3 to code  iso
remove unneeded flowof in country handler
adding message source to country handler
deleting the sql from the country repository
deleting the unneeded method in country repository
adding annotation to country router for  swagger ui
creating country router info for swagger annotation
fixing country router test
adding remover method from the repository to the service
adding tests in country service test
changing the FraudsterAndCountryCodeDTO to data class
moving the sonar params to the maven yml file
adding more messages to messages.properties
deleting the @Autowired annotation
removing the mono and refactoring the code of PersonDatabaseRepository.kt
adding new tests to the PersonDatabaseRepositoryTest.kt
changing the PersonDTO.kt to data class and removing the builder name
adding missing errors to the PersonHandler.kt
creating PersonInCountry.kt data class
changing the PersonsByCountry.kt number of user to int
refactoring person service adding specific errors
linting PersonServiceTest.kt
removing sonar config from pom
changing the code3 to code3 iso in schema.sql
adding OpenAPIDefinition to main class

ImedZnd committed:
changing the name of the queue listener and the queue, we can handel multi consumer
removing name builder from country dao
refactor country database repository
refactor country handler and add  getAllPersonInCountry
refactor country reactive repository
adding allPersonInAllCountry Route in country router
adding allPersonInAllCountry to swagger annotation
refactor country service and add countryCodeToCountPersonByCountry method
creating package for each DTO to respect the DDD
create LocalDateDTO.kt
adding BadPersonIdInPath message
changing the year to int in person
creating personQueuePublisher to separate form the repository
refactor person config and add personQueuePublisher  method
 change the year to int in person dao
 refactor person database repository and delete the queuing methods
 refactor person dto
 refactor person handler
 delete person id dto
 delete person id dto
 refactor person queue listener
 refactor person repository and delete publishing methods
 refactor person router using the get in path variable simple values
 refactor person annotation swagger ui
 refactor person service
 delete year dto
 change year dto to handle int instead of year
linting class country, country config, tests

ImedZnd committed:
removing the Persistable interface from country dao and the id annotation
adding find by code to country reactive interface
change the find by id to find by code
add spacing to local date dto
linting person database repository and year range dto

ImedZnd committed:
adding sleeping time when using queue person test service
